addon/components/ember-tether.js

```diff
@@ -1,140 +1,102 @@
 import { schedule } from '@ember/runloop';
-import { computed, get, observer } from '@ember/object';
 import { isNone } from '@ember/utils';
-import Component from '@ember/component';
+import Component from '@glimmer/component';
 import Tether from 'tether';
+import { action } from '@ember/object';
 
-export default Component.extend({
-  classNames: ['ember-tether'],
-  classPrefix: 'ember-tether',
-  target: null,
-  attachment: null,
-  targetAttachment: null,
-  offset: null,
-  targetOffset: null,
-  targetModifier: null,
-  constraints: null,
-  optimizations: null,
-  emberTetherConfig: computed(function() {
-    return (getOwner(this).resolveRegistration('config:environment') || {})['ember-tether'];
-  }),
-  bodyElement: computed(function() {
-    let config = get(this, 'emberTetherConfig');
+export default class EmberTetherComponent extends Component {
+  _tether;
+  element;
+
+  get classPrefix() {
+    return this.args.classPrefix || 'ember-tether';
+  }
+
+  get emberTetherConfig() {
+    return (getOwner(this).resolveRegistration('config:environment') || {})[
+      'ember-tether'
+    ];
+  }
+
+  get bodyElement() {
+    let config = this.emberTetherConfig;
     if (config && config.bodyElementId) {
       return document.getElementById(config.bodyElementId);
     }
-  }),
-  attributeBindings: [
-    'aria-atomic',
-    'aria-busy',
-    'aria-controls',
-    'aria-current',
-    'aria-describedby',
-    'aria-details',
-    'aria-disabled',
-    'aria-errormessage',
-    'aria-flowto',
-    'aria-haspopup',
-    'aria-hidden',
-    'aria-invalid',
-    'aria-keyshortcuts',
-    'aria-label',
-    'aria-labelledby',
-    'aria-live',
-    'aria-owns',
-    'aria-relevant',
-    'aria-roledescription'
-  ],
-  didInsertElement() {
-    this._super(...arguments);
-    this.addTether();
-  },
+    return undefined;
+  }
 
-  willDestroyElement() {
-    this._super(...arguments);
+  willDestroy() {
+    super.willDestroy(...arguments);
     if (!this._tether) return;
-
     let { _tether, element } = this;
     schedule('render', () => {
       this.removeElement(element);
       this.removeTether(_tether);
     });
-  },
-
-  didRender() {
-    this._super(...arguments);
-    this.positionTether();
-  },
+  }
 
-  tetherDidChange: observer(
-    'classPrefix',
-    'target',
-    'attachment',
-    'targetAttachment',
-    'offset',
-    'targetOffset',
-    'targetModifier',
-    'constraints',
-    'optimizations',
-    function() {
-      this.removeTether(this._tether);
-      this.addTether();
-    }
-  ),
+  @action updateTether() {
+    this.removeTether(this._tether);
+    this.addTether();
+  }
 
+  @action
   positionTether() {
-    if (this._tether) {
-      this._tether.position();
-    }
-  },
+    this._tether?.position();
+  }
 
-  addTether() {
-    if (get(this, '_tetherTarget')) {
-      this._tether = new Tether(this._tetherOptions());
+  @action
+  addTether(el = null) {
```

Comment:
Not a blocker, but this would be a great candidate for a custom modifier that manages the tether, in the future.